packages/web/test/mocks/myalgowallet-mock.ts

```diff
@@ -0,0 +1,91 @@
+import type {
+    Options,
+    Accounts,
+    Address,
+    AlgorandTxn,
+    Base64,
+    ConnectionSettings,
+    EncodedTransaction,
+    SignedTx,
+    SignTransactionOptions,
+} from "@randlabs/myalgo-connect";
+
+export default class MyAlgoConnectMock {
+
+    /**
+     * @param {Options} options Override default popup options.
+     */
+    constructor(options?: Options) {
+
+    };
+
+    /**
+     * @async
+     * @description Receives user's accounts from MyAlgo.
+     * @param {ConnectionSettings} [settings] Connection settings
+     * @returns Returns an array of Algorand addresses.
+     */
+    connect(settings?: ConnectionSettings): Promise<Accounts[]> {
+        return new Promise((resolve, reject) => {
+            return resolve([{
+                address: "",
+                name: ""
+            }]);
+        });
+    };
+
+    /**
+     * @async
+     * @description Sign an Algorand Transaction.
+     * @param transaction Expect a valid Algorand transaction
+     * @param signOptions Sign transactions options object.
+     * @returns Returns signed transaction
+     */
+    signTransaction(transaction: AlgorandTxn | EncodedTransaction, signOptions?: SignTransactionOptions): Promise<SignedTx>;
+
+    /**
+     * @async
+     * @description Sign an Algorand Transaction.
+     * @param transaction Expect a valid Algorand transaction array.
+     * @param signOptions Sign transactions options object.
+     * @returns Returns signed an array of signed transactions.
+     */
+    signTransaction(transaction: (AlgorandTxn | EncodedTransaction)[], signOptions?: SignTransactionOptions): Promise<SignedTx[]>;
+
+    signTransaction(transaction: AlgorandTxn | EncodedTransaction | (AlgorandTxn | EncodedTransaction)[], signOptions?: SignTransactionOptions): Promise<SignedTx | SignedTx[]> {
```

Comment:
Not sure if i follow. 2 declarations?

Reply:
yeah there official docs also have the two similar method declarations since the parameters type is different and return type is also different

Reply:
alright